src/pysoem/pysoem.pyx

```diff
@@ -239,6 +241,13 @@ cdef class CdefMaster:
             ret_val = cpysoem.ecx_init_redundant(&self._ecx_contextt, &self._ecx_redport, ifname.encode('utf8'), ifname_red.encode('utf8'))
         if ret_val == 0:
             raise ConnectionError('could not open interface {}'.format(ifname))
+
+        self.context_initialized = True
+
+    def __check_context_is_initialized(self):
+        if not self.context_initialized:
+            raise EnvironmentError("Soem Context is not initialized or has been closed. Call Master.open() first")
```

Comment:
I'm not sure if the EnvironmentError exception type is the right fit. 
The python [docs](https://docs.python.org/3/library/exceptions.html#EnvironmentError) say: 
>The following exceptions are kept for compatibility with previous versions; starting from Python 3.3, they are aliases of [OSError](https://docs.python.org/3/library/exceptions.html#OSError).
>
>exception EnvironmentError

If something like this is run:
```py
with open("foo.txt") as f:
    pass

f.read()
```
Python throws an ValueError. (`ValueError: I/O operation on closed file.`)
Also not very fitting...
What about the [UnboundLocalError](https://docs.python.org/3/library/exceptions.html#UnboundLocalError) I already used at some places:
>exception UnboundLocalError
>
>    Raised when a reference is made to a local variable in a function or method, but no value has been bound to that variable. This is a subclass of [NameError](https://docs.python.org/3/library/exceptions.html#NameError).

Reply:
You are right, the exception type is not that clear. I think the best choice is a custom exception, like the other exceptions that are thrown on pysoem